File: DBFlow-Compiler/src/main/java/com/raizlabs/android/dbflow/processor/validator/Validator.java
package com.raizlabs.android.dbflow.processor.validator;

import com.raizlabs.android.dbflow.processor.model.ProcessorManager;

/**
 * Description: the base interface for validating annotations.
 */
public interface Validator<ValidatorDefinition> {

    /**
     * @param processorManager    The manager
     * @param validatorDefinition The validator to use
     * @return true if validation passed, false if there was an error.
     */
    public boolean validate(ProcessorManager processorManager, ValidatorDefinition validatorDefinition);
}

File: dbflow-processor/src/main/java/com/raizlabs/android/dbflow/processor/validator/Validator.kt
package com.raizlabs.android.dbflow.processor.validator

import com.raizlabs.android.dbflow.processor.model.ProcessorManager

/**
 * Description: the base interface for validating annotations.
 */
interface Validator<in ValidatorDefinition> {

    /**
     * @param processorManager    The manager
     * *
     * @param validatorDefinition The validator to use
     * *
     * @return true if validation passed, false if there was an error.
     */
    fun validate(processorManager: ProcessorManager, validatorDefinition: ValidatorDefinition): Boolean
}
